src/components/DroneStatus/MPSControlSection.tsx

```diff
@@ -1,94 +1,83 @@
-import { Button, Grid, Stack, TextField } from "@mui/material";
+import { Button, Grid, TextField } from "@mui/material";
 import { useState } from "react";
 import { armDrone, disarmDrone, takeoffDrone } from "../../api/droneEndpoints";
-import ErrorSnackbar, { SnackbarState } from "../ErrorSnackbar";
+import { openSnackbar } from "../../store/slices/appSlice";
+import { useAppDispatch } from "../../store/store";
 
 export default function MPSControlSection() {
-    const [snackbarState, setSnackbarState] = useState<SnackbarState>({
-        message: "",
+    const [controlState, setControlState] = useState({
         armed: false,
         takeoffAltitude: 0,
     });
+    const dispatch = useAppDispatch();
 
     const handleArming = async (arming: boolean) => {
         try {
             arming ? await armDrone() : await disarmDrone();
         } catch (error) {
             const message = (error as Error).message;
             console.log(message);
-            setSnackbarState({ message, armed: true });
+            dispatch(openSnackbar(message));
         }
     };
 
     const updateSnackBarState = async (event: React.ChangeEvent<HTMLInputElement>) => {
         if (["altitude"].includes(event.target.id) && /[^0-9.-]/.test(event.target.value)) {
             return;
         }
-        setSnackbarState({
-            ...snackbarState,
-            [event.target.id]: event.target.value,
-        });
+        setControlState({ ...controlState, takeoffAltitude: parseFloat(event.target.value) });
     };
 
     const handleTakeoff = async (altitude?: number) => {
         try {
-            if (!snackbarState.armed) {
+            if (!controlState.armed) {
                 console.log("Please arm the drone.");
             }
             await takeoffDrone(altitude);
         } catch (error) {
             const message = (error as Error).message;
             console.log(message);
-            setSnackbarState({ message, armed: true });
```

Comment:
This was very strange, since `armed: true` is needed to open the snackbar but also allows takeoff to actually run since `armed` is now `true`. This is one of the main reasons I opted to refactor.